src/osg/Image.cpp

```diff
@@ -241,11 +241,17 @@ Image::Image(const Image& image,const CopyOp& copyop):
     {
         unsigned int size = image.getTotalSizeInBytesIncludingMipmaps();
         setData(new unsigned char [size],USE_NEW_DELETE);
-        unsigned char* dest_ptr = _data;
-        for(DataIterator itr(&image); itr.valid(); ++itr)
+        if (unsigned char* dest_ptr = _data) 
```

Comment:
How far back in C++ does putting variable declarations in if statements go?  It seems like a bit of risk for breaking the build on older systems.

Reply:
There's a new feature in C++17 that allows you to put multiple declarations in one if statement, but the simple `if (type var = ....)` has always been here as far as I know. I never had an issue with that.

And by the way, this syntax is already used in OpenSceneGraph [here](https://github.com/openscenegraph/OpenSceneGraph/blob/3b1212fed59b1024aab289bf7837056c559913e1/examples/osgmultiplemovies/osgmultiplemovies.cpp#L599).

Reply:
Thanks for spotting the existing usage, tells us it should be safe to merge :-)